crates/sui-types/src/messages.rs

```diff
@@ -512,17 +512,25 @@ pub struct TransactionEnvelope<S> {
     #[serde(skip)]
     pub is_verified: bool,
 
-    pub data: TransactionData,
-    /// tx_signature is signed by the transaction sender, applied on `data`.
-    pub tx_signature: Signature,
+    // The packet of data that authorities will sign on. Stores the tx data and the sender signature.
+    pub signed_data: SignedData,
+
     /// authority signature information, if available, is signed by an authority, applied on `tx_signature` || `data`.
     pub auth_sign_info: S,
     // Note: If any new field is added here, make sure the Hash and PartialEq
     // implementation are adjusted to include that new field (unless the new field
     // does not participate in the hash and comparison).
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SignedData {
+    pub data: TransactionData,
+    /// tx_signature is signed by the transaction sender, applied on `data`.
+    pub tx_signature: Signature,
+}
+
 impl<S> TransactionEnvelope<S> {
+    #[allow(dead_code)]
```

Comment:
Why is that not used?

Reply:
If we commit this, then there is no longer a reason to add user signatures to the VerificationObligation batcher, as authorities and users are signing over different messages.

Reply:
We can probably delete this function when we are committed to merging this PR